pkg/i2gw/provider.go

```diff
@@ -42,7 +42,8 @@ type ProviderConstructor func(conf *ProviderConf) Provider
 // ProviderConf contains all the configuration required for every concrete
 // Provider implementation.
 type ProviderConf struct {
-	Client client.Client
+	Client  client.Client
+	Options ProviderImplementationSpecificOptions
```

Comment:
Why do we need the `Options` field here?
From what I've seen, each provider constructs its own `ProviderImplementationSpecificOptions` and passes it to `ToGateway` (which is good, in my opinion).
So I don't see the purpose of having it here too

Reply:
😕 we don't indeed. I added such a field when thinking about Lior's comments and accidentally committed. Just removed, good catch